### PR TITLE
splunk-audit-exporter: expose NonSystemChangeValidationWebhookConfiguration as metric

### PR DESCRIPTION
**Pull request by typeid**
**What?**
This PR updates `splunk-audit-exporter` to the latest version which allows exposing selected events as metric using the policy file. 

Along with updating the `splunk-audit-exporter` version, this PR adds the following objects to allow HTTPs scraping:
- ServiceMonitor
- RoleBinding for prometheus
- Role for prometheus

The daemonset for the `splunk-audit-exporter` is extended to receive TLS cert & key and to open up port 9090.

The namespace `openshift-security` namespace also receives an additional label to allow prometheus scraping.

**Why?**

This PR updates `splunk-audit-exporter` to expose the equivalent of the `Non-System Change ValidatingWebhookConfigurations` webhook set in splunk as metric within the cluster (implemented in [OSD-10551](https://issues.redhat.com/browse/OSD-10551)). 

This particular metric will be used in [OSD-13558](https://issues.redhat.com/browse/OSD-13558) and [OSD-13559](https://issues.redhat.com/browse/OSD-13559) to trigger automated service logs for the event corresponding to a `NonSystemChangeValidatingWebhookConfiguration`.  

**Validation**

Tested by applying all updated ressources as below to a stage cluster, and verifying that the metric gets exposed and scraped when editing validation webhooks:
`oc edit validatingwebhookconfigurations/sre-pod-validation --as backplane-cluster-admin`

### CODE REVIEW

**Review by mjlshen — APPROVED** [review]
/lgtm

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -992,4 +1005,72 @@ objects:
...
+      metadata:
+        name: audit-exporter
+        namespace: openshift-security
+      spec:
+        endpoints:
+          - path: /metrics
+            port: metrics
+            scheme: https
+            tlsConfig:
+              caFile: /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
```
Comment (mjlshen) [inline hack/olm-registry/olm-artifacts-template.yaml:1049]: TIL - Thanks for this, it is a really great idea, ref: https://docs.openshift.com/container-platform/4.11/security/certificate_types_descriptions/service-ca-certificates.html

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 2]
```diff
@@ -546,14 +547,16 @@ objects:
...
               - --log-file=/var/log/osd-audit/audit.log
+              - --tls-cert-file=/certs/tls.crt
+              - --tls-private-key-file=/certs/tls.key
               resources:
                 requests:
                   cpu: 10m
                   memory: 32Mi
                 limits:
                   cpu: 50m
                   memory: 128Mi
```
Comment (gsleeman) [inline hack/olm-registry/olm-artifacts-template.yaml:558]: Can we bump these up a little, just to be safe? (metrics aren't enabled at all in prod currently) 

Maybe make the requests & limits the same to ensure the pods get a QoS class of Guaranteed

```suggestion
              resources:
                requests:
                  cpu: 100m
                  memory: 256Mi
                limits:
                  cpu: 100m
                  memory: 256Mi
```

**Review by gsleeman — CHANGES_REQUESTED** [review]

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 2 above)
Comment (typeid) [reply hack/olm-registry/olm-artifacts-template.yaml]: Sounds good.

**Pull request merged**